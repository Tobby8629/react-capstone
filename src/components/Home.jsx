import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';

function Home() {
  const [date, setdate] = useState('');
  const data = useSelector((state) => state.Data.info);

  const searchfilter = (e) => {
    const { value } = e.target;
    setdate(value);
  };

  return (

    <>
      <div className="header">
        <h2>Global Apple product Revenue</h2>
      </div>
      <div className="home">
        <div className="form">
          <input type="text" placeholder="filter by year example 2022-09-03" onChange={searchfilter} />
        </div>
        {data.filter((info) => info.date.includes(date)).map((data) => (
          <div className="each" key={data.id}>
            <h3>{data.date}</h3>
            <div className="detail">
              <p>
                Total revenue globally:
                {data.total}
              </p>
              <Link to={`card/${data.id}`}> details </Link>
            </div>
          </div>
        ))}
      </div>
    </>

  );
}

export default Home;
