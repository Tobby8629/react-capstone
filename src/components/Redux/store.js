import { configureStore } from "@reduxjs/toolkit";
import dataReducer from './data/dataSlice'

const store = configureStore({
    reducer: {
      Data: dataReducer,
    }
})

export default store