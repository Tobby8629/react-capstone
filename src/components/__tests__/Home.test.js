import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../../Redux/store';
import Home from '../Home';

test('Home match snapshot', () => {
  const tree = renderer.create(<Provider store={store}><Home /></Provider>);
  expect(tree).toMatchSnapshot();
});
