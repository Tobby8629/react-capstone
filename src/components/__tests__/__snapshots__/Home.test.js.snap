// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home match snapshot 1`] = `
Array [
  " ",
  <div
    className="home"
  >
    <div
      className="form"
    >
      <input
        onChange={[Function]}
        placeholder="filter by year"
        type="text"
      />
    </div>
  </div>,
]
`;
